subsys/bluetooth/host/Kconfig

```diff
@@ -4,14 +4,29 @@
 # Copyright (c) 2015-2016 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
+config BT_LONG_WQ
+	# Enable dedicated wq for long-running tasks
+	bool "Enable dedicated workqueue for long-running tasks."
+	default y if BT_GATT_CACHING || BT_TINYCRYPT_ECC
+
+if BT_LONG_WQ
+config BT_LONG_WQ_STACK_SIZE
+	int "Long workqueue stack size."
```

Comment:
I get what you mean, but the same can be argued for `BT_HCI_ECC_STACK_SIZE` which is public.

I made it settable so quick fixes would be possible without source changes. I don't have a strong opinion though, I can make it private if you want.

Reply:
We could have the `BT_HCI_ECC_STACK_SIZE` `default` on some decent values depending on whether ECC/GATT caching is enabled. I assume that if we have GATT caching without ECC it can be smaller?

Reply:
@Thalley that's what I was going for/expecting. Or maybe I misunderstand your comment ?

Reply:
No, I think we are aligned: Make this private and have the stack size depend on what internal requirements we have (as the long_wq isn't available to the application).

Reply:
The one argument I can think of is if the stack size may vary depending on the architecture and compilation options (e.g. if optimization is enabled), however I think there might be some system-wide option affect all thread stacks for such cases?